tests/test_reply_function_async.js

```diff
@@ -89,3 +90,35 @@ test('reply should throw on error on the callback', t => {
 
   req.end()
 })
+
+test('an error passed to the callback when a full response array is expected', async t => {
+  const scope = nock('http://example.test')
+    .get('/')
+    .reply((path, requestBody, callback) => {
+      callback(Error('boom'))
+    })
+
+  await assertRejects(got('http://example.test/'), ({ statusCode, body }) => {
+    t.is(statusCode, 500)
+    t.matches(body, 'Error: boom')
+    return true
+  })
+
+  scope.done()
+})
+
+test('subsequent calls to the reply callback are ignored', async t => {
+  const scope = nock('http://example.com')
+    .get('/')
+    .reply(201, (path, requestBody, callback) => {
+      callback(null, 'one')
+      callback(null, 'two')
+      callback(null, 'three')
```

Comment:
Should we add a `t.ok()` here with a `t.plan()` to make sure we get all the way through this function before the test finishes?

Reply:
Added.